src/actions/wallet/L1/writeUnsafeDepositTransaction.ts

```diff
@@ -33,33 +36,46 @@ export type WriteUnsafeDepositTransactionParameters<
     TChainOverride
   >,
   'abi' | 'functionName' | 'args' | 'address'
-> & {
-  toChain: OpChainL2
+> & ({
+  toChainId: ExtractValidChainIdFromContract<_resolvedChain, _contractName>
+  optimismPortal: never,
   args: DepositTransactionParameters
-}
+} | {
+  toChainId: never,
+  optimismPortal: Address
+  args: DepositTransactionParameters
+})
+
+type ExtractValidChainIdFromContract<TChain extends OpStackL1Chain | undefined, contractName extends keyof OpStackL1Chain['contracts']> = TChain extends OpStackL1Chain ? keyof TChain['contracts'][contractName] : undefined
+type ResolveChain<TChain extends Chain | undefined, TChainOverride extends Chain | undefined = undefined> = TChainOverride extends Chain ? TChainOverride : TChain
 
 export async function writeUnsafeDepositTransaction<
-  TChain extends Chain | undefined,
+  TChain extends OpStackL1Chain | undefined,
   TAccount extends Account | undefined,
-  TChainOverride extends Chain | undefined,
+  TChainOverride extends OpStackL1Chain | undefined,
 >(
-  client: WalletClient<Transport, TChain, TAccount>,
+  client: WalletClient<Transport, OpStackL1Chain, TAccount>,
```

Comment:
This is required for L68, otherwise complains client.chain maybe not defined. Again not sure if best way to do it